kernel/syscall.c

```diff
@@ -516,6 +516,30 @@ static bool _iov_bad_addr(const struct iovec* iov, int iovcnt)
     return false;
 }
 
+static int _brk_pids_set(uint64_t new_brk)
+{
+    int ret = 0;
+    static uint64_t old_brk = 0;
+
+    if (!old_brk)
```

Comment:
Does the context guarantee that `_brk_pids_set` will always be called with appropriate locks?
If yes, it is better to add a comment reflecting that.
If no, then `_brk_pids_set` needs to modify `old_brk` in a thread-safe manner.

Reply:
brk is not supported by this PR